content/06.discussion.md

```diff
@@ -54,6 +54,34 @@ In assessing the upper bound on the predictive performance of a deep learning mo
+
+#### Confidence calibration, confidence intervals, and out-of-distribution detection
+
+In classification tasks, confidence calibration is the problem of using classifier scores to predict class membership probabilities that match the true membership likelihoods. These membership probabilities can be used to assess the uncertainty associated with assigning the example to each of the classes. Guo et al [@tag:guo_calibration] observed that contemporary neural networks are poorly calibrated and provide a simple recommendation for calibration: temperature scaling, a single parameter special case of Platt scaling [@tag:platt_scaling]. In addition to confidence calibration, there is early work from Chryssolouris et al [@tag:Chryssolouris1996_confidence] that described a method for obtaining confidence intervals with the assumption of normally distributed error for the neural network. More recently, Hendrycks and Gimpel discovered that incorrect or out-of-distribution examples usually have lower maximum softmax probabilities than correctly classified examples, allowing for effective detection of misclassified examples [@tag:out_dist_baseline]. Liang et al used temperature scaling and small perturbations to further separate the softmax scores of correctly classified examples and the scores of out-of-distribution examples, allowing for more effective detection [@tag:temp_out_dist].
+
+#### Bayesian neural networks
+
+An alternative approach for obtaining principled uncertainty estimates from deep learning models is to use Bayesian neural networks. Deep learning models are usually trained to obtain the most likely parameters given the data. However, choosing the single most likely set of parameters ignores the uncertainty about which parameters are best and sometimes leads to uncertainty in predictions when the chosen likely parameters produce high-confidence but incorrect results. On the other hand, the parameters of Bayesian neural networks are modeled as full probability distributions. This Bayesian approach comes with a whole host of safety benefits, including: better calibrated confidence estimates [@tag:ai_safety] and more robustness to adversarial and out-of-distribution examples [@tag:strong_adversary]. Unfortunately, modeling the full posterior distribution for the model’s parameters given the data is usually computationally intractable. One popular method for circumventing this high computational cost is called test-time dropout and can be found in Gal and Ghahramani [@tag:Gal2015_dropout], where an approximate posterior distribution is obtained using variational inference.
```

Comment:
Minor complaint, but I don't like "best" as a word - it's a bit fuzzy.

Reply:
I feel like "safety" benefits comes out of left field here. I suggest either removing the term or else you need to explain it. You could say "This Bayesian approach comes with a whole host of benefits including better calibrated confidence estimates [@tag:ai_safety] and more robustness to adversarial and out-of-distribution examples [@tag:strong_adversary]."

Reply:
I clarified what I meant by "best" and removed the word "safety".